packages/inventory/src/loaders/connection.ts

```diff
@@ -1,14 +1,23 @@
-import { ConfigModule } from "@medusajs/medusa"
+import {
+  ConfigModule,
+  ConfigurableModuleDeclaration,
+  LoaderOptions,
+  MODULE_RESOURCE_TYPE,
+} from "@medusajs/medusa"
 import { ConnectionOptions, createConnection } from "typeorm"
 import { CONNECTION_NAME } from "../config"
 
 import { ReservationItem, InventoryItem, InventoryLevel } from "../models"
 
-export default async ({
-  configModule,
-}: {
-  configModule: ConfigModule
-}): Promise<void> => {
+export default async (
+  { configModule }: LoaderOptions,
+  moduleDeclaration?: ConfigurableModuleDeclaration
+): Promise<void> => {
+  // Do not connect to the database if using shared resources with the Core
+  if (moduleDeclaration?.resources === MODULE_RESOURCE_TYPE.SHARED) {
```

Comment:
thought: right now the module would not work without being in a shared scenario so I am wondering if we should remove the connection loader altogether first time around?

Reply:
My point is that we should leave a potential code path open that might not work. Wdyt?

Reply:
That's a good point.
I'll throw an Error at the constructor if the resource is not shared until we have eventServices as a Module